x/bank/types/key.go

```diff
@@ -60,6 +60,9 @@ func CreateAccountBalancesPrefix(addr []byte) []byte {
 // CreateDenomAddressPrefix creates a prefix for a reverse index of denomination
 // to account balance for that denomination.
 func CreateDenomAddressPrefix(denom string) []byte {
-	key := append(DenomAddressPrefix, []byte(denom)...)
-	return append(key, 0)
+	key := make([]byte, len(DenomAddressPrefix)+len(denom)+1)
+	copy(key, DenomAddressPrefix)
+	copy(key[len(DenomAddressPrefix):], denom)
+	// key[last] = 0 - not needed, because this is the default
```

Comment:
why are we actually adding a 0 at the end? (in the initial design)

Reply:
I believe it's for null terminating and prefix iterating. Right @aaronc?

Reply:
correct, the denom has variable length. I will add a comment in the code.